lib/workers/repository/process/write.ts

```diff
@@ -40,10 +47,33 @@ export async function writeUpdates(
     }
     addMeta(meta);
     const branchExisted = branchExists(branch.branchName);
-    const res = await processBranch(branch);
+    let branchCache = {} as BranchCache;
+    if (branchExisted && config.repositoryCache === 'enabled') {
+      branchCache =
+        cachedBranches?.find((br) => br.branchName === branch.branchName) ??
+        ({} as BranchCache);
+
+      if (Object.keys(branchCache).length === 0) {
+        logger.debug(`No branch cache found for ${branch.branchName}`);
+      }
+    }
+    const branchManagersFingerprint = hasha(
+      branch.upgrades
+        .map((upgrade) => hashMap.get(upgrade.manager) ?? upgrade.manager)
+        .filter(is.string)
+    );
+    branch.branchFingerprint = hasha([
+      JSON.stringify(branch),
+      branchManagersFingerprint,
+    ]);
+
+    const res = await processBranch(branch, branchCache);
```

Comment:
At this point you already know (a) cached fingerprint, and (b) current fingerprint. I don't think we need to pass the info through in such a complicated manner, including the whole branchCache unless it's used for other things.

Reply:
yes it isn't necessary to pass branchCache. I will remove the branchCache parameter, and shift the `canSkipBranchUpdateCheck` function to write.ts. I think we can pass the result (can skip or not) using a new config property like `config.skipBranchUpdate`.